code/game/objects/structures/syndicate_uplink_beacon.dm

```diff
@@ -0,0 +1,115 @@
+/// Device that traitors can craft in order to be sent a new, undisguised uplink
+/obj/structure/syndicate_uplink_beacon
+	name = "suspicious beacon"
+	icon = 'icons/obj/machines/telecomms.dmi'
+	icon_state = "relay_traitor"
+	desc = "This ramshackle device seems capable of recieving and sending signals for some nefarious purpose."
+	density = TRUE
+	anchored = TRUE
+	/// Traitor's code that they speak into the radio
+	var/uplink_code
+	/// weakref to person who is going to use the beacon to get a replacement uplink
+	var/datum/weakref/owner
+	/// while constructed the teleport beacon is still active
+	var/obj/item/beacon/teleport_beacon
+	/// Radio that the device needs to listen to the codeword from the traitor
+	var/obj/item/radio/listening_radio
+	/// prevents traitor from activating teleport_beacon proc too much in a small period of time
+	COOLDOWN_DECLARE(beacon_cooldown)
+
+/obj/structure/syndicate_uplink_beacon/Initialize(mapload)
+	. = ..()
+	register_context()
+
+	var/static/list/tool_behaviors = list(
+		TOOL_SCREWDRIVER = list(
+			SCREENTIP_CONTEXT_RMB = "Deconstruct",
+		),
+	)
+	AddElement(/datum/element/contextual_screentip_tools, tool_behaviors)
+	listening_radio = new(src)
+	listening_radio.canhear_range = 0
+	teleport_beacon = new(src)
+
+/obj/structure/syndicate_uplink_beacon/attack_hand(mob/living/user, list/modifiers)
+	if(!IS_TRAITOR(user))
+		balloon_alert(user, "don't know how to use!")
+		return
+	if(IS_WEAKREF_OF(owner, user))
+		balloon_alert(user, "already synchronized to you!")
+		return
+	if(owner != null)
+		balloon_alert(user, "already claimed!")
+		return
+	var/datum/looping_sound/typing/typing_sounds = new(src, /*start_immediately =*/TRUE)
+	balloon_alert(user, "synchronizing...")
+	if(!do_after(user = user, delay = 3 SECONDS, target = src, interaction_key = REF(src)))
+		typing_sounds.stop()
+		return
+	typing_sounds.stop()
+	probe_traitor(user)
+
+/obj/structure/syndicate_uplink_beacon/screwdriver_act_secondary(mob/living/user, obj/item/tool)
+	tool.play_tool_sound(src)
+	balloon_alert(user, "deconstructing...")
+	if (!do_after(user, 5 SECONDS, target = src))
+		return FALSE
+	var/turf/beacon_tile = get_turf(src)
+	new /obj/item/stack/sheet/iron/five(beacon_tile)
+	new /obj/item/stack/cable_coil/five(beacon_tile)
+	teleport_beacon.forceMove(beacon_tile)
+	teleport_beacon = null
```

Comment:
don't we do this in qdel(src) anyways? keep in mind calling `qdel()` on a null value (which is what `QDEL_NULL` will do when we invoke it the uplink_beacon's `Destroy()`) will provoke it to call `del()` on a null value on line 67 which should cause a bad del runtime IIRC.

you always want to follow the order of operatures set out in `QDEL_NULL`, which is to qdel the item and immediately set the variable to null (which should be governed by the macro anyways).

Reply:
what i was doing was having the beacon that pops out of the machine not get qdel'd due to its association to the machine when the machine gets qdel'd

what ive done now is have destroy not qdel teleport_beacon if there is no teleport beacon inside of the machine which should only happen if the machine was deconstructed and spat the teleport beacon out